test/celeritas/MockTestBase.cc

```diff
@@ -223,8 +224,9 @@ auto MockTestBase::build_physics() -> SPConstPhysics
         inp.materials = this->material();
         inp.interact = interact;
         inp.label = "barks";
-        inp.applic = {make_applicability("electron", 1e-5, 10)};
+        inp.applic = {make_applicability("electron", 1e-5, 1e9)};
```

Comment:
The test failure in `PhysicsStepUtilsTest.calc_mean_energy_loss`:
```console
celeritas/phys/PhysicsStepUtils.hh:268:
celeritas: internal assertion failed: range > step" thrown in the test body.
```
happens with a CGS build too depending on what this upper applicable energy is set to. For some values it passes with:
```console
range=2.0200000000000010e-03, step=2.0200000000000001e-03, eloss=1.0100000000000000e-03, pre_step_energy=1.0000000000000000e-03
```
and for others it fails with:
```console
range=2.0199999999999966e-03, step=2.0200000000000001e-03, eloss=1.0100000000000000e-03, pre_step_energy=1.0000000000000000e-03
```

Reply:
Um. Can we tell where the discrepancy is occurring? I thought we were strict about this even accounting for floating point arithmetic 🤔

Reply:
Just a testing issue I think. The value of the range can change slightly depending on the energy bounds or number of grid points, and in this case it looks like it became slightly less than the step; in `calc_physics_step_limit()` the step limit would have been set equal to the range so we wouldn't have hit that assertion, but this test only calls `calc_mean_energy_loss()`.

I'm assuming this test is meant to test [this case](https://github.com/celeritas-project/celeritas/blob/20d2ba457817d76f607573de757d140c9cc877d1/src/celeritas/phys/PhysicsStepUtils.hh#L270-L272)...